### PR TITLE
Add JSDoc to public methods

### PR DESCRIPTION
**Pull request by xy1xy**
1. Add JSDocs to public methods on `seq_logger.js`
2. Fix up some typing errors on `index.d.ts`
3. _ESLint_ did it's thing - I can revert these formatting changes if you wish

### CODE REVIEW

File: seq_logger.js [hunk 1]
```diff
@@ -80,27 +83,34 @@ class SeqLogger {
...
         if (this._closed) {
             throw new Error('The logger has already been closed.');
         }
-        
+
         this._closed = true;
         this._clearTimer();
         return this.flush().then(() => {
             this._httpAgent.destroy();
         });
```
Comment (xy1xy) [inline seq_logger.js:105]:
```suggestion
        return this.flush().then(() => {
            return this._httpAgent.destroy(); // <-- should this be returned? other, `close` returns `Promise<void>`
        });
```

File: index.d.ts [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 export declare class Logger {
-  constructor(config: SeqLoggerConfig);
+  constructor (config: SeqLoggerConfig)
 
-  emit(event: SeqEvent): void;
-  close(): Promise<boolean>;
-  flush(): Promise<boolean>;
-  flushToBeacon(): Promise<boolean>;
+  emit (event: SeqEvent): void
+  close (): Promise<void>
```
Comment (xy1xy) [inline index.d.ts:32]: `close` actually returns _Promise<void>_

File: index.d.ts [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
+  flush (): Promise<boolean>
+  flushToBeacon (): boolean
```
Comment (xy1xy) [inline index.d.ts:34]: `flushToBeacon` does not return a _Promise<boolean>_, just a _boolean_

File: seq_logger.js [hunk 4]
```diff
@@ -50,18 +50,21 @@ class SeqLogger {
...
     }
 
-    // Flush events queued at the time of the call, and wait for pending writes to complete
-    // regardless of configured batching/timers.
-    flush() {
+    /**
+     * Flush events queued at the time of the call, and wait for pending writes to complete regardless of configured batching/timers.
+     * @returns {Promise<boolean}
+     */
+    flush () {
```
Comment (xy1xy) [inline seq_logger.js:57]: JSDoc can be seen in the intellisense:
image

File: seq_logger.js [hunk 5]
```diff
@@ -50,18 +50,21 @@ class SeqLogger {
...
-    // Events over 63K in length are discarded (with a warning sent in its place) 
-    // and the total size batch will be no more than 63K in length.
-    flushToBeacon() {
+    /**
+     * * A browser only function that queues events for sending using the navigator.sendBeacon() API.    
+     * * This may work in an unload or pagehide event handler when a normal flush() would not.    
+     * * Events over 63K in length are discarded (with a warning sent in its place) and the total size batch will be no more than 63K in length.
+     * @returns {boolean} 
+     */
+    flushToBeacon () {
```
Comment (xy1xy) [inline seq_logger.js:67]: Same here:
image

**Comment by xy1xy** [comment]
Bump :)

**Comment by nblumhardt** [comment]
Sorry about the slow response- will check this out today 👍

File: seq_logger.js (hunk 1 above)
Comment (nblumhardt) [reply seq_logger.js:105]: As far as I can tell, `destroy()` returns `void` - so probably not, here

**Comment by nblumhardt** [comment]
LGTM!

In a small codebase like this I'm okay with the ESLint defaults :-) 👍

**Comment by xy1xy** [comment]
I forgot to regen the definition file, so the JSDocs are exposed :facepalm:

**Comment by xy1xy** [comment]
> I forgot to regen the definition file, so the JSDocs are exposed facepalm

Fixup PR: https://github.com/datalust/seq-logging/pull/47

**Pull request merged**